alchemist/alchemist-cognitive-agents/src/main/kotlin/it/unibo/alchemist/model/cognitiveagents/characteristics/cognitive/AbstractCognitiveCharacteristic.kt

```diff
@@ -6,6 +6,13 @@ import it.unibo.alchemist.model.cognitiveagents.characteristics.PARAMETERS_FILE
 abstract class AbstractCognitiveCharacteristic : CognitiveCharacteristic {
 
     protected var currentLevel: Double = 0.0
+        set(value) {
+            field = when {
```

Comment:
isn't this a very counterintuitive way for `min(1.0, max(0.0, value))`? by the way, there is probably a library function to constrain values